packages/@lwc/integration-karma/test/component/observed-fields/index.spec.js

```diff
@@ -133,6 +135,57 @@ describe('observed-fields', () => {
         });
     });
 
+    it('should rerender component when computed field with spaces is mutated', () => {
+        const elm = createElement('x-computed', { is: Computed });
+        document.body.appendChild(elm);
+
+        expect(elm.getValue('with spaces')).toBe('spaces!');
+
+        elm.setValue('with spaces', 'mutated');
+
+        return Promise.resolve().then(() => {
+            expect(elm.shadowRoot.querySelector('.spaces').textContent).toBe('mutated');
+        });
+    });
+
+    it('should rerender component when computed field with number is mutated', () => {
+        const elm = createElement('x-computed', { is: Computed });
+        document.body.appendChild(elm);
+
+        expect(elm.getValue(1337)).toBe('number!');
+
+        elm.setValue(1337, 'mutated');
+
+        return Promise.resolve().then(() => {
+            expect(elm.shadowRoot.querySelector('.number').textContent).toBe('mutated');
+        });
+    });
+
+    it('should rerender component when computed field with symbol is mutated', () => {
+        const elm = createElement('x-computed', { is: Computed });
+        document.body.appendChild(elm);
+
+        expect(elm.getValue(Computed.symbol)).toBe('symbol!');
+
+        elm.setValue(Computed.symbol, 'mutated');
+
+        return Promise.resolve().then(() => {
+            expect(elm.shadowRoot.querySelector('.symbol').textContent).toBe('mutated');
+        });
+    });
+
+    it('should rerender component when computed field with prefixed field is mutated', () => {
+        const elm = createElement('x-computed', { is: Computed });
+        document.body.appendChild(elm);
+
+        expect(elm.getValue(`${Computed.prefix}Field`)).toBe('prefixed!');
+        elm.setValue(`${Computed.prefix}Field`, 'mutated');
+
+        return Promise.resolve().then(() => {
+            expect(elm.shadowRoot.querySelector('.prefixed').textContent).toBe('mutated');
+        });
+    });
```

Comment:
I included template literals for discussion purposes. It's not in the original issue and should probably be removed before merging.

Reply:
The trouble with template literals, and all computed expressions, is that they can change.

```js
class extends LightningElement {
  [`${Date.now()}`] = 'now'
}
```

In the compiled output, we currently just copy the content of the computed property, i.e. we have `` `${Date.now()}` `` twice. Which means that it's entirely possible that the timestamp in the class definition is a millisecond off from the timestamp in the call to `_registerDecorators`. That's definitely no good!

As described in https://github.com/salesforce/lwc/issues/3956#issuecomment-1901089168, to use computed properties in a more predictable manner, we'll likely need to compute the properties once and store the result. But, like you said, that'll definitely be outside this PR.

Reply:
Also, fwiw, [TypeScript doesn't like](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBBCeBbARiANjAvDAyk1aAFAEQAWAhhcQJQDcAUKJLAAoBKAogGICSAGlhjEADgCcApgDMAlgA9iDJtBhipcrtPFoAJoIAGAEgDe7bvwC+Grdr31xs4SFGxtU8gFc0sYGnIQIMAAq4spG9DAwANrE5MQAuoIADAwRkQCMAMwZAOwJ2MnhUQgo6HlCxQQAhAqFkaoyslY6ZcQgANbk8DWphiacvHyWmjoATHotYCCw7Z3dUXrI7rBQpNIBazAz8OOCAOQAju7Som3wuwzm9EA) computed template literals directly in computed props.

Reply:
Thanks! I thought of that exact case, but still wanted to confirm